src/System.Runtime.CompilerServices.Unsafe/tests/UnsafeTests.cs

```diff
@@ -459,6 +459,24 @@ public static void RefAdd()
         }
 
         [Fact]
+        public static unsafe void VoidPointerAdd()
+        {
+            int[] a = new int[] { 0x123, 0x234, 0x345, 0x456 };
+
+            fixed (void* ptr = a)
+            {
+                void* r1 = Unsafe.Add<int>(ptr, 1);
```

Comment:
since you are operating on native pointers, please add more test cases around edge cases